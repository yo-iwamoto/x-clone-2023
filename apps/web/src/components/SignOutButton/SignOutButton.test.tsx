import { SignOutButton } from '.';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const { signOutActionMock, routerRefreshMock } = vi.hoisted(() => ({
  signOutActionMock: vi.fn(),
  routerRefreshMock: vi.fn(),
}));
vi.mock('./action', () => ({
  signOutAction: signOutActionMock,
}));
vi.mock('next/navigation', () => ({
  useRouter: vi.fn(() => ({
    refresh: routerRefreshMock,
  })),
}));

describe('SignOutButton', () => {
  it('snapshot unchanged', () => {
    const { asFragment } = render(<SignOutButton />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('signOutAction を呼び出すこと', async () => {
    const { getByRole } = render(<SignOutButton />);

    await userEvent.click(getByRole('button', { name: 'Sign out' }));

    expect(signOutActionMock).toHaveBeenCalled();
  });

  it('サインアウト後、Router Cache が refresh されること', async () => {
    const { getByRole } = render(<SignOutButton />);

    await userEvent.click(getByRole('button', { name: 'Sign out' }));

    expect(routerRefreshMock).toHaveBeenCalled();
  });
});
