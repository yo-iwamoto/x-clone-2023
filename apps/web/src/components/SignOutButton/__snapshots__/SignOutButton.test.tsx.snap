// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignOutButton > snapshot unchanged 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
    type="button"
  >
    Sign out
  </button>
</DocumentFragment>
`;
